config-linux.md

```diff
@@ -158,13 +158,20 @@ cgroups provide controls to restrict cpu, memory, IO, pids and network for the c
 For more information, see the [kernel cgroups documentation][cgroup-v1].
 
 The path to the cgroups can be specified in the Spec via `cgroupsPath`.
-`cgroupsPath` is expected to be relative to the cgroups mount point.
-If `cgroupsPath` is not specified, implementations can define the default cgroup path.
+If `cgroupsPath` is
+* ... an absolute path (starting with `/`), the implementation MUST take the path to be relative to the cgroup mount point.
+* ... a relative path (not starting with `/`), implementations SHOULD take the value to be the suffix of the resulting cgroup path.
+* ... not specified, implementations CAN define the default cgroup path.
+Implementations CAN consider certain `cgroupsPath` values to be invalid, and MUST generate an error if this is the case.
+
 Implementations of the Spec can choose to name cgroups in any manner.
 The Spec does not include naming schema for cgroups.
 The Spec does not support [split hierarchy][cgroup-v2].
 The cgroups will be created if they don't exist.
 
+A runtime MUST at least use the minimum set of cgroups required to fulfil the resource restrictions specified by `config.json`.
+However, a runtime CAN decide to not attach to all cgroups supported by the system.
```

Comment:
@vishh The devices cgroup thing is honestly an implementation detail of `runC`. Since users can drop `CAP_MKNOD` (and use user namespaces) there isn't a security issue with runtimes not joining devices -- as far as I can see (users can always set the device restrictions for the container anyway).

>  How about the case when no resource restrictions are specified?

Then the runtime doesn't need to join any cgroups. However, we should probably add an "update" step to the lifecycle so that we can specify that a runtime MUST join any cgroups required to fulfill the resource restriction updates (using the same `cgroupsPath` rules as with setup).